### PR TITLE
Use kelseyhightower/envconfig pkg to validate the environment variables

### LINKED ISSUE #2659: Use kelseyhightower/envconfig pkg to validate the environment variables.
Currently, we have a lot of env vars in the litmusportal, so we need to use `envconfig` pkg to validate and process the env vars.
PKG- https://github.com/kelseyhightower/envconfig

Need to add in:
- graphql-server
- event-tracker
- subscriber
- authentication-server

### PR DESCRIPTION
**Pull request by 123vivekr**
Signed-off-by: Vivek R <123vivekr@gmail.com>

## Proposed changes

Currently, we have a lot of env vars in the litmusportal, so this PR uses envconfig pkg to validate and process the env vars.
PKG- https://github.com/kelseyhightower/envconfig

Modules changed: 
- graphql-server
- event-tracker
- subscriber
- authentication-serve

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

## Special notes for your reviewer:
Closes #2659

### CODE REVIEW

**Review by rajdas98 — APPROVED** [review]

**Comment by rajdas98** [comment]
Screenshot 2021-09-27 at 1 39 39 PM

Works fine for me

File: litmus-portal/graphql-server/server.go [hunk 1]
```diff
@@ -28,14 +29,41 @@ import (
...
+	AgentScope                  string `required:"true" split_words:"true"`
+	AgentNamespace              string `required:"true" split_words:"true"`
+	LitmusPortalNamespace       string `required:"true" split_words:"true"`
+	DbUser                      string `required:"true" split_words:"true"`
+	DbPassword                  string `required:"true" split_words:"true"`
+	PortalScope                 string `required:"true" split_words:"true"`
+	SubscriberImage             string `required:"true" split_words:"true"`
+	EventTrackerImage           string `required:"true" split_words:"true"`
+	ArgoWorkfloWControllerImage string `required:"true" split_words:"true"`
+	ArgoWorkflow_ExecutorImage  string `required:"true" split_words:"true"`
```
Comment (rajdas98) [inline litmus-portal/graphql-server/server.go:48]:
```suggestion
	ArgoWorkflowExecutorImage  string `required:"true" split_words:"true"`
```

**Review by rajdas98 — CHANGES_REQUESTED** [review]

**Comment by gdsoumya** [comment]
Similar changes are needed for [upgrade agent](https://github.com/litmuschaos/litmus/tree/master/litmus-portal/upgrade-agents/control-plane) too

**Review by ishangupta-ds — APPROVED** [review]
LGTM

**Review by rajdas98 — APPROVED** [review]

**Review by gdsoumya — APPROVED** [review]

**Comment by rajdas98** [comment]
Thanks, @123vivekr for this PR

**Pull request merged**